### PR TITLE
Device data Android Unity changes

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Adds public accessors to DeviceData and DeviceDataSummary, for use by 3rd party developers and internal API (denoted with marker annotation).

Note: the intention is for further work to be carried out in a separate PR (e.g. migrating legacy fields to metadata, extracting the data capture to a separate class, etc)

### Changed

Converted fields/methods to properties (getter/setter backed with a field) in DeviceData and DeviceDataSummary.

## Tests

Adapted unit tests for JSON serialisation of each class, and added a test for the default value + override value of each accessor.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: sdk/src/main/java/com/bugsnag/android/DeviceData.java [hunk 1]
```diff
@@ -70,46 +83,115 @@ public void toStream(@NonNull JsonStream writer) throws IOException {
...
+        metaData.addToTab("device", "brand", Build.BRAND);
+        metaData.addToTab("device", "apiLevel", Build.VERSION.SDK_INT);
+        metaData.addToTab("device", "osBuild", Build.DISPLAY);
+        metaData.addToTab("device", "locale", locale);
+        metaData.addToTab("device", "screenDensity", screenDensity);
+        metaData.addToTab("device", "dpi", dpi);
+        metaData.addToTab("device", "emulator", isEmulator());
+        metaData.addToTab("device", "screenResolution", screenResolution);
+        metaData.addToTab("device", "cpuAbi", cpuAbi);
+    }
```
Comment (martin308) [inline sdk/src/main/java/com/bugsnag/android/DeviceData.java]: How were you thinking that the unity notifier would access all of these properties?

Comment (fractalwrench) [reply sdk/src/main/java/com/bugsnag/android/DeviceData.java]: For the Android notifier, these need to be added to `error.metadata`. My thinking is that we could expose an internal method on `Client` that allows for the population of this metadata. Something like:

```
